File: java/src/model/logger/Logger.java
package model.logger;

public class Logger {

    private Nivel nivel = null;

    private static Logger instance = null;

    private Logger(Nivel nivel) {
        this.nivel = nivel;
    }

    public static Logger getInstance(Nivel nivel) {
        if (instance == null) {
            instance = new Logger(nivel);
        }
        return instance;
    }

    public static Logger getInstance() {
        if (instance == null) {
            instance = new Logger(Nivel.INFO);
            Logger.getInstance().info("El Logger ha sido instanciado a nivel INFO por defecto");
        }
        return instance;
    }

    public Nivel getNivel() {
        return nivel;
    }

    public void setNivel(Nivel nivel) {
        this.nivel = nivel;
    }

    public void debug(String respuesta){
        System.out.println("DEBUG: " + respuesta);
    }

    public void info(String respuesta){
        if (getNivel() != Nivel.DEBUG){
            System.out.println("INFO: " + respuesta);
        }
    }

    public void warn(String respuesta){
        if( getNivel() != Nivel.DEBUG || getNivel() != Nivel.INFO){
            System.out.println("WARN: " + respuesta);
        }
    }

    public void error(String respuesta){
        if(getNivel() == Nivel.ERROR){
            System.out.println("ERROR: " + respuesta);
        }
    }
}

File: python/model/logger/logger.py
from model.logger.nivel import Nivel

class Logger():

    _instance = None

    def __init__(self):
        raise("Call instance(Nivel value) to create Logger")

    @classmethod
    def instance(cls, nivel=Nivel.INFO):
        if cls._instance is None:
            cls._instance = cls.__new__(cls)
            cls._nivel = nivel
        return cls._instance

    @property
    def nivel(self):
        return self._nivel

    @nivel.setter
    def nivel(self, nivel):
        self._nivel = nivel

    def debug(self, respuesta):
        print("DEBUG: " + respuesta)

    def info(self, respuesta):
        if(self.nivel != Nivel.DEBUG):
            print("INFO: " + respuesta)

    def warn(self, respuesta):
        if( self.nivel != Nivel.DEBUG or self.nivel == Nivel.INFO):
            print("WARN: " + respuesta)

    def error(self, respuesta):
        if( self.nivel == Nivel.ERROR):
            print("ERROR: " + respuesta)
        
